dowhy/causal_refuters/linear_sensitivity_analysis.py

```diff
@@ -0,0 +1,435 @@
+        """
+
+        t_alpha_df_1 = t.ppf(self.significance_level / 2 , self.degree_of_freedom - 1) # t-value threshold with alpha significance level and dof-1 degrees of freedom
+        critical_t = abs(t_alpha_df_1) * np.sign(self.t_stats)
+
+        contour_values = []
+        for x in r2tu_w:
+            contour = []
+            for y in r2yu_tw:
+                benchmarking_results = self.compute_bias_adjusted(r2tu_w = x, r2yu_tw = y)
+                t_value = benchmarking_results['bias_adjusted_t']
+                contour.append(t_value)
+            contour_values.append(contour)
+
+        t_bounds = self.bias_adjusted_t
+        return contour_values, critical_t, t_bounds
+
+    
+    def plot(self, sensitivity_variable = 'estimate', critical_estimate = 0, critical_t = 2, contour_color = "blue", threshold_color = "red", x_limit = 0.8, y_limit = 0.8 ):
+        """
+        The horizontal axis shows hypothetical values of the partial R2 of unobserved confounder(s) with the treatment
+        The vertical axis shows hypothetical values of the partial R2 of unobserved confounder(s) with the outcome.
+        The contour levels represent adjusted t-values or estimates for unobserved confounders with these hypothetical partialR2 values.
+        We also plot bounds on the partial R^2 of the unobserved confounders obtained from observed covariates
+
+        :param sensitivity_variable: "estimate" or "t-value"
+        :param critical_estimate: threshold line to represent the contour corresponding to estimate, (default = 0)
+        :param critical_t: threshold line to represent the contour corresponding to t-value, (default = 2 - usual approx value for 95% CI)
+        :param x_limit: plot's maximum x_axis value
+        :param y_limit: plot's minimum y_axis value
+        :param contour_color: color of contour line(default = blue)
+        :param threshold_color: color of threshold line(default = red)
+        """
+
+        #Plotting the contour plot
+        fig, ax = plt.subplots(1, 1, figsize = (7,7))
+        ax.set_title("Sensitivity contour plot of %s"  %sensitivity_variable)
+        ax.set_xlabel("Partial R^2 of confounder with treatment")
+        ax.set_ylabel("Partial R^2 of confounder with outcome")
+
+        for i in range(len(self.r2tu_w)):
+            x = self.r2tu_w[i]
+            y = self.r2yu_tw[i]
+            if(x > 0.8 or y > 0.8):
+                x_limit = 0.95
+                y_limit = 0.95
+                break
+
+        r2tu_w = np.arange(0.0, x_limit, x_limit / 200)
+        r2yu_tw = np.arange(0.0, y_limit, y_limit / 200)
+
+        unadjusted_point_estimate = None
+
+        if sensitivity_variable == "estimate":
+            contour_values, contour_threshold, bound_values  = self.plot_estimate(r2tu_w, r2yu_tw)
+            unadjusted_estimate = self.estimate
+            unadjusted_point_estimate = unadjusted_estimate
+        else:

```

Comment:
still good to check elif t-value. Else return a ValueError saying that we only support estimate or t-value as params